pkg/certificate/mrc_reconciler.go

```diff
@@ -1,214 +1,212 @@
-	// TODO(5046): determine the status to set the owned components to.
-	var status v1alpha2.MeshRootCertificateComponentStatus
-
-	for _, useCase := range m.ownedUseCases {
-		switch useCase {
-		case UseCaseValidatingWebhook:
-			// validating, issuing unknown.
-			mrc.Status.ComponentStatuses.ValidatingWebhook = status
-		case UseCaseMutatingWebhook:
-			mrc.Status.ComponentStatuses.MutatingWebhook = status
-		case UseCaseXDSControlPlane:
-			mrc.Status.ComponentStatuses.XDSControlPlane = status
-		case UseCaseSidecar:
-			mrc.Status.ComponentStatuses.Sidecar = status
-		case UseCaseBootstrap:
-			mrc.Status.ComponentStatuses.Bootstrap = status
-		case UseCaseGateway:
-			mrc.Status.ComponentStatuses.Gateway = status
+	for _, intent := range validIntents {
+		if intent2 == intent {
+			log.Debug().Msgf("verified valid intent combination of %s intent and %s intent", intent1, intent2)
+			return nil
 		}
 	}
 
-	return m.mrcClient.UpdateMeshRootCertificate(mrc)
+	log.Error().Err(ErrInvalidMRCIntentCombination).Str(errcode.Kind, errcode.GetErrCodeWithMetric(errcode.ErrInvalidMRCIntentCombination)).
+		Msgf("invalid combination of %s intent and %s intent", intent1, intent2)
+	return ErrInvalidMRCIntentCombination
 }
 
-// isTerminal checks if the MRC is in a terminal state.
-func isTerminal(mrc *v1alpha2.MeshRootCertificate) bool {
-	// TODO(5046): check if the mrc is in a terminal state.
-	return true
-}
+// TODO(jeallio): simplify to different data structure rather than user switch case
+// or pass intent directly to set issuers. It has already be
+func (m *Manager) setIssuers(mrc1, mrc2 *v1alpha2.MeshRootCertificate) error {
+	if mrc1 == nil || mrc2 == nil {
+		log.Error().Err(ErrMRCNotFound).Msg("cannot validate nil mrc")
+		return ErrMRCNotFound
+	}
 
-// state and condition get updated together.
-func (m *Manager) updateMRCState(mrc *v1alpha2.MeshRootCertificate) error {
-	// TODO(5046): update the MRC state & conditions, and issue the update via the MRCClient.
+	issuer1, err := m.getCertIssuer(mrc1)
+	if err != nil {
+		return err
+	}
+	issuer2, err := m.getCertIssuer(mrc2)
+	if err != nil {
+		return err
+	}
 
-	// If it's not in a terminal state, set the next transition time.
-	// NOTE: consider that this isTerminal check will need to apply against the updated mrc state which should happen
-	// above in this method.
-	if isTerminal(mrc) {
-		mrc.Status.TransitionAfter = nil
-	} else {
-		mrc.Status.TransitionAfter = &metav1.Time{Time: time.Now().Add(mrcDurationPerStage)}
+	intent1 := mrc1.Spec.Intent
+	intent2 := mrc2.Spec.Intent
+
+	var signingIssuer, validatingIssuer *issuer
+	switch intent1 {
+	case v1alpha2.ActiveIntent:
+		switch intent2 {
+		case v1alpha2.PassiveIntent:
+			signingIssuer = issuer1
+			validatingIssuer = issuer2
+		case v1alpha2.InactiveIntent:
+			signingIssuer = issuer1
+			validatingIssuer = issuer1
+		case v1alpha2.DeactiveIntent:
```

Comment:
After going through this as an end user, having a `Deactivated` intent but it is still being used felt a bit funny, would it be possible to mark as passive here?   I think we got rid of `inactive` case and `deactivate` case then the switch becomes flat instead of nested which would make this section easier to understand (though this isn't too bad)

I think the challenge would be that we wouldn't know which one to set as Validator an signer in a deterministic way? Is that important or could we use oldest cert or some other field?

Reply:
Discussed offline. Removed `Deactive`/`Deactivated` intent.